pkg/virtualization/core/virtualsystem/virtualmachine.go

```diff
@@ -674,7 +674,9 @@ func (vm *VirtualMachine) NewGpuPartition(partitionSizeBytes uint64) (newGpuPart
 		return
 	}
 
-	err = newGpuPartitionSettingData.SetPropertyMinPartitionVRAM(partitionSizeBytes)
+	if partitionSizeBytes > 0 {
```

Comment:
instead, can you add a check to return error if partitionSize is 0

Reply:
For hybrid aks, since we only support homogeneous environment, windows expert for gpu kernel suggests us to leave the parameter as empty. So for hybrid aks, we will leave partitionsize as 0. In such case, we will not set the partition size.

Reply:
I dont think its a good idea to design these APIs based on the product. We shouldnt assume in this layer of code.

In this method, I see that input i partitionSize passed in and by principle, we validate and throw if the input is not correct.
If there are scenarios where empty partitionSize is passed, please implement another method to not pass in that param and use it for your purpose.